pkg/event/target/amqp.go

```diff
@@ -166,7 +166,7 @@ func NewAMQPTarget(id string, args AMQPArgs) (*AMQPTarget, error) {
 		if err == nil {
 			break
 		}
-		if err != nil && i == 5 {
+		if i == 5 {
```

Comment:
How is this a tautological error?

Reply:
`err` is guaranteed to not equal `nil`, if it were we would have triggered the `break` above.

Reply:
It should be restructured such that the error case is handled first and nil should break after. That is a correct fix here. Otherwise it's complex to understand the way it's modified now.

Reply:
Fair enough, I inverted the error handling and pushed.